lib/livebook_web/plugs/auth_plug.ex

```diff
@@ -12,7 +12,18 @@ defmodule LivebookWeb.AuthPlug do
   @impl true
   def call(conn, _opts) do
     if authenticated?(conn) do
-      conn
+      user =
+        LivebookWeb.UserPlug.build_current_user(
+          get_session(conn),
+          conn.assigns.identity_data,
+          conn.assigns.user_data
+        )
+
+      if user.restricted_apps_groups == nil do
+        conn
+      else
+        render_unauthorized(conn)
+      end
```

Comment:
I would leave a comment for clarity:

```suggestion
      if user.restricted_apps_groups do
        # User has access only to specific app pages, so they do not have
        # access to any pages guarded by this plug.
        render_unauthorized(conn)
      else
        conn
      end
```